src/render.ts

```diff
@@ -100,35 +100,64 @@ function extractResources(
   )
 }
 
+function queryChanges(changes: StructuredPlanChange[], changeKind: string): string[] {
+  return changes
+    .filter((r) => r.change.actions.toString() === [changeKind].toString())
+    .map((r) => r.address)
+}
+
+function getRecreatedChanges(changes: StructuredPlanChange[]): string[] {
+  return changes
+    .filter(
+      (r) =>
+        r.change.actions.toString() === ['delete', 'create'].toString() ||
+        r.change.actions.toString() === ['create', 'delete'].toString()
+    )
+    .map((r) => r.address)
+}
+
 export function internalRenderPlan(
   structuredPlan: StructuredPlanfile,
   humanReadablePlan: string
 ): RenderedPlan {
   // If there are no changes, we do not need to build any sections
   if (
-    structuredPlan.resource_changes === undefined ||
-    structuredPlan.resource_changes.length === 0
+    (
+      structuredPlan.resource_changes === undefined ||
+      structuredPlan.resource_changes.length === 0
+    ) && (
+      structuredPlan.resource_drift === undefined ||
+      structuredPlan.resource_drift.length === 0
+    )
   ) {
     return {}
   }
 
   // Partition changes for output formatting and extract resources
-  const createdResources = structuredPlan.resource_changes
-    .filter((r) => r.change.actions.toString() === ['create'].toString())
-    .map((r) => r.address)
-  const updatedResources = structuredPlan.resource_changes
-    .filter((r) => r.change.actions.toString() === ['update'].toString())
-    .map((r) => r.address)
-  const recreatedResources = structuredPlan.resource_changes
-    .filter(
-      (r) =>
-        r.change.actions.toString() === ['delete', 'create'].toString() ||
-        r.change.actions.toString() === ['create', 'delete'].toString()
-    )
-    .map((r) => r.address)
-  const deletedResources = structuredPlan.resource_changes
-    .filter((r) => r.change.actions.toString() === ['delete'].toString())
-    .map((r) => r.address)
+  const createdResources = queryChanges(
+    structuredPlan.resource_changes || [],
+    'create'
+  ).concat(queryChanges(
+    structuredPlan.resource_drift || [],
+    'create',
+  ))
```

Comment:
Can we create a new function which takes care of the concatenation s.t. we don't have to do this multiple times?